From 289faa0a181b33119086945baaaa4f1e364801fd Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <uwu@icenowy.me>
Date: Tue, 16 Jan 2024 19:33:20 +0800
Subject: [PATCH] fix NULL dereference on wxWidgets 3.2

According to the document of wxWidgets, the initializer of wxLocale with
no parameters do not do any initialization at all, thus do not register
its wxTranslations.

Explicitly call Init() with its default parameters to init it with the
default setting, otherwise on Linux with wxWidgets 3.2, the
wxTranslations singleton get is NULL and lead to NULL deference.

The code here is from PrusaSlicer 2.5 [1].

[1] https://github.com/prusa3d/PrusaSlicer/commit/e21921f2ebd2b94c57459bd280f1b1bd8eec1b9c

Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
---
 src/slic3r/GUI/GUI_App.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/slic3r/GUI/GUI_App.cpp b/src/slic3r/GUI/GUI_App.cpp
index d10c5cfe0..6dc647dc0 100644
--- a/src/slic3r/GUI/GUI_App.cpp
+++ b/src/slic3r/GUI/GUI_App.cpp
@@ -2289,6 +2289,15 @@ bool GUI_App::load_language(wxString language, bool initial)
         {
 	    	// Allocating a temporary locale will switch the default wxTranslations to its internal wxTranslations instance.
 	    	wxLocale temp_locale;
+#ifdef __WXOSX__
+            // ysFIXME - temporary workaround till it isn't fixed in wxWidgets:
+            // Use English as an initial language, because of under OSX it try to load "inappropriate" language for wxLANGUAGE_DEFAULT.
+            // For example in our case it's trying to load "en_CZ" and as a result PrusaSlicer catch warning message.
+            // But wxWidgets guys work on it.
+            temp_locale.Init(wxLANGUAGE_ENGLISH);
+#else
+            temp_locale.Init();
+#endif // __WXOSX__
 	    	// Set the current translation's language to default, otherwise GetBestTranslation() may not work (see the wxWidgets source code).
 	    	wxTranslations::Get()->SetLanguage(wxLANGUAGE_DEFAULT);
 	    	// Let the wxFileTranslationsLoader enumerate all translation dictionaries for PrusaSlicer
-- 
2.39.1

